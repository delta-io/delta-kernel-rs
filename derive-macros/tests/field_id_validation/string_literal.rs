use delta_kernel_derive::ToSchema;

// Dummy types to avoid dependency issues in test
pub trait ToSchema {
    fn to_schema() -> StructType;
}

pub struct StructType;

#[derive(ToSchema)]
struct TestStruct {
    #[field_id = "123"]  // Should be integer, not string
    field: String,
}

fn main() {}
