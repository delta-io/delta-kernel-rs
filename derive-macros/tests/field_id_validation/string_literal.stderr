error: field_id must be an integer literal, not a string
  --> tests/field_id_validation/string_literal.rs:12:18
   |
12 |     #[field_id = "123"]  // Should be integer, not string
   |                  ^^^^^

error[E0433]: failed to resolve: use of unresolved module or unlinked crate `delta_kernel`
  --> tests/field_id_validation/string_literal.rs:10:10
   |
10 | #[derive(ToSchema)]
   |          ^^^^^^^^ use of unresolved module or unlinked crate `delta_kernel`
   |
   = help: if you wanted to use a crate named `delta_kernel`, use `cargo add delta_kernel` to add it to your `Cargo.toml`
   = note: this error originates in the derive macro `ToSchema` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0433]: failed to resolve: use of unresolved module or unlinked crate `delta_kernel`
  --> tests/field_id_validation/string_literal.rs:10:10
   |
10 | #[derive(ToSchema)]
   |          ^^^^^^^^ use of unresolved module or unlinked crate `delta_kernel`
   |
   = help: if you wanted to use a crate named `delta_kernel`, use `cargo add delta_kernel` to add it to your `Cargo.toml`
   = note: this error originates in the derive macro `ToSchema` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing this struct
   |
1  + use crate::StructType;
   |
